resteasy/src/main/java/com/linecorp/armeria/client/resteasy/ArmeriaResteasyClientBuilder.java

```diff
@@ -0,0 +1,541 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.resteasy;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetSocketAddress;
+import java.security.KeyStore;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Configuration;
+
+import org.jboss.resteasy.client.jaxrs.ClientHttpEngine;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.ClientFactoryBuilder;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.client.proxy.ProxyConfig;
+import com.linecorp.armeria.common.SessionProtocol;
+
+/**
+ * An optional helper class to build {@link ResteasyClient} using {@link ResteasyClientBuilder} interface
+ * as below.
+ * <pre>{@code
+ *     final WebClientBuilder webClientBuilder = WebClient.builder(); // supply no server URI to the builder
+ *     // ... configure webClientBuilder ...
+ *     final ResteasyClientBuilder reasteasyClientBuilder =
+ *             ArmeriaResteasyClientBuilder.newBuilder(webClientBuilder);
+ *     // ... configure restClientBuilder ...
+ *     // construct JAX-RS client
+ *     final Client jaxrsClient = reasteasyClientBuilder.build();
+ *     // construct JAX-RS web target
+ *     final WebTarget webTarget = jaxrsClient.target(restServerUri); // supply server URI here
+ *     // make JAX-RS request
+ *     final Response restResponse = webTarget.path(servicePath).request().get();
+ * }</pre>
+ *
+ * <p>
+ * {@link ResteasyClient} could still be constructed using ArmeriaJaxrsClientEngine directly by setting it
+ * to {@link ResteasyClientBuilder} via {@link ResteasyClientBuilder#httpEngine(ClientHttpEngine)} method
+ * as below.
+ * <pre>{@code
+ *     final Client jaxrsClient = ((ResteasyClientBuilder) ClientBuilder.newBuilder())
+ *             .httpEngine(new ArmeriaJaxrsClientEngine(armeriaWebClient))
+ *             .build();
+ * }</pre>
+ * </p>
+ */
+public final class ArmeriaResteasyClientBuilder extends ResteasyClientBuilder {
+
+    /**
+     * Creates new {@link ResteasyClientBuilder} based on {@link WebClientBuilder}.
+     */
+    public static ResteasyClientBuilder newBuilder(WebClientBuilder webClientBuilder) {
```

Comment:
I think we can just add a setter which sets `Consumer<WebClientBuilder>` instead accepting `webClientBuilder` at constructor because it's not mandatory to build Armeria `ResteasyClient`.
```java
public ArmeriaResteasyClientBuilder armeriaClientCustomizer(Consumer<WebClientBuilder> customizer) {...}
```
```

Reply:
I wanted to make this helper class as generic as possible and remove any need to cast to concrete `ArmeriaResteasyClientBuilder` so that existing client code using RESTEasy API or JAX-RS API won't be impacted so much. That's why you see `ResteasyClientBuilder newBuilder(WebClientBuilder webClientBuilder)`, instead of `ArmeriaResteasyClientBuilder newBuilder(WebClientBuilder webClientBuilder)`.
In fact, as mentioned in the JavaDoc, this builder is more like a convenience class. It should be still possible to construct the client using just the Armeria engine directly as following:
```
     final Client jaxrsClient = ((ResteasyClientBuilder) ClientBuilder.newBuilder())
             .httpEngine(new ArmeriaJaxrsClientEngine(armeriaWebClient))
             .build();
```